deepchem/models/wandblogger.py

```diff
@@ -0,0 +1,115 @@
+                 metrics: List[Metric],
+                 train_dataset_id: Optional[Union[str, int]] = None,
+                 name: Optional[str] = None,
+                 entity: Optional[str] = None,
+                 project: Optional[str] = None,
+                 save_dir: Optional[str] = None,
+                 mode: Optional[str] = "online",
+                 id: Optional[str] = None,
+                 anonymous: Optional[str] = "never",
+                 log_model: Optional[bool] = False,
+                 **kwargs):
+        try:
+            import wandb
+        except ImportError:
+            raise ImportError(
+                'You want to use `wandb` logger which is not installed yet,'
+                ' install it with `pip install wandb`.'
+            )
+
+        if mode == "offline" and log_model:
+            raise Exception(
+                f'Providing log_model={log_model} and mode={mode} is an invalid configuration'
+                ' since model checkpoints cannot be uploaded in offline mode.\n'
+                'Hint: Set `mode="online"` to log your model.'
+            )
+        #Check for datasets and metrics
+        if (datasets is None) or (not datasets):
+            raise Exception(
+                'No datasets are provided: `datasets` is not set or is empty.'
+            )
+        if (metrics is None) or (not metrics):
+            raise Exception(
+                'No metrics are provided: `metrics` is not set or is empty'
+            )
+
+
+        self.base_model = None # will be set in KerasModel init
+        self.datasets = datasets
+        #if datasets is in list format, convert to dictionary
+        if isinstance(datasets, list):
+            self.datasets = {"dataset_" + str(k): v for k, v in enumerate(self.datasets)}
+
+        #identify the train dataset for epoch calculation purposes
+        # assign first dataset as train set by default
+        if isinstance(train_dataset_id, int) and isinstance(datasets, list):
+            self.train_dataset_name = "dataset_" + str(train_dataset_id)
+        elif isinstance(train_dataset_id, str) and isinstance(datasets, dict):
+            self.train_dataset_name = train_dataset_id
+        else:
+            self.train_dataset_name = list(self.datasets.keys())[0]
+            logger.warning("WandbLogger: 'train_dataset_id' not provided or invalid. "
+                           "It must be a str/int for the dataset at key/index of 'datasets'."
+                           "Train dataset will default to first dataset in 'datasets'.")
+
+        self.metrics = metrics
+
+        self.log_model = log_model
+        self.save_dir = save_dir
+
+        # set wandb init arguments
+        self.wandb_init_params = dict(
+            name=name,
+            project=project,
+            entity=entity,
+            mode=mode,
+            id=id,
+            dir=save_dir,
+            resume='allow',
```

Comment:
We might want to make this a parameter to the wandbcallback, just for any rare cases people might want a different option